linkerd/service-profiles/src/split.rs

```diff
@@ -58,31 +61,67 @@ impl<N: Clone, S, Req> Clone for NewSplit<N, S, Req> {
     }
 }
 
-impl<T, N: Clone, S, Req> NewService<T> for NewSplit<N, S, Req>
+impl<T, N, S, Req> NewService<T> for NewSplit<N, S, Req>
 where
-    T: AsRef<Receiver>,
+    T: Clone,
+    for<'t> &'t T: Into<Addr> + Into<Option<Receiver>>,
+    N: NewService<(Option<Addr>, T), Service = S> + Clone,
     S: tower::Service<Req>,
+    S::Error: Into<Error>,
 {
     type Service = Split<T, N, S, Req>;
 
     fn new_service(&mut self, target: T) -> Self::Service {
-        let rx = target.as_ref().clone();
-        Split {
-            rx,
-            target,
-            new_service: self.inner.clone(),
-            rng: self.rng.clone(),
-            inner: None,
-            services: ReadyCache::default(),
-        }
+        let inner = match Into::<Option<Receiver>>::into(&target) {
```

Comment:
nit/tioli: personally i might have written this with an early return in the `None` case instead of putting most of the code inside the match, like 
```rust
let rx = match Into::<Option<Receiver>>::into(&target) {
    Some(rx) => rx,
    None => {
        trace!("Building default service");
        return Self { inner: Inner::Default(self.inner.new_service((None, target))) };
    },
};

let mut targets = rx.borrow().targets.clone();
// ...
```
but i'm not actually sure if this is any clearer...